optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/main/java/org/optaplanner/quarkus/deployment/OptaPlannerProcessor.java

```diff
@@ -211,31 +214,77 @@ private void applySolverProperties(RecorderContext recorderContext,
     }
 
     private void applyScoreDirectorFactoryProperties(IndexView indexView, SolverConfig solverConfig) {
+        Optional<String> constraintsDrlFromProperty = constraintsDrl();
+        Optional<String> defaultConstraintsDrl = defaultConstraintsDrl();
+        Optional<String> effectiveConstraintsDrl = constraintsDrlFromProperty.map(Optional::of).orElse(defaultConstraintsDrl);
         if (solverConfig.getScoreDirectorFactoryConfig() == null) {
-            ScoreDirectorFactoryConfig scoreDirectorFactoryConfig = new ScoreDirectorFactoryConfig();
-            scoreDirectorFactoryConfig.setEasyScoreCalculatorClass(
-                    findImplementingClass(DotNames.EASY_SCORE_CALCULATOR, indexView));
-            scoreDirectorFactoryConfig.setConstraintProviderClass(
-                    findImplementingClass(DotNames.CONSTRAINT_PROVIDER, indexView));
-            scoreDirectorFactoryConfig.setIncrementalScoreCalculatorClass(
-                    findImplementingClass(DotNames.INCREMENTAL_SCORE_CALCULATOR, indexView));
+            ScoreDirectorFactoryConfig scoreDirectorFactoryConfig =
+                    defaultScoreDirectoryFactoryConfig(indexView, effectiveConstraintsDrl);
+            solverConfig.setScoreDirectorFactoryConfig(scoreDirectorFactoryConfig);
+        } else {
+            ScoreDirectorFactoryConfig scoreDirectorFactoryConfig = solverConfig.getScoreDirectorFactoryConfig();
+            if (constraintsDrlFromProperty.isPresent()) {
+                scoreDirectorFactoryConfig.setScoreDrlList(Collections.singletonList(constraintsDrlFromProperty.get()));
+            } else {
+                if (scoreDirectorFactoryConfig.getScoreDrlList() == null) {
+                    defaultConstraintsDrl.ifPresent((resolvedConstraintsDrl) -> scoreDirectorFactoryConfig
+                            .setScoreDrlList(Collections.singletonList(resolvedConstraintsDrl)));
+                }
+            }
+        }
+
+        if (solverConfig.getScoreDirectorFactoryConfig().getScoreDrlList() != null) {
             ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
-            if (classLoader.getResource(SolverBuildTimeConfig.DEFAULT_SCORE_DRL_URL) != null) {
-                scoreDirectorFactoryConfig.setScoreDrlList(Collections.singletonList(
-                        SolverBuildTimeConfig.DEFAULT_SCORE_DRL_URL));
+            try {
+                Class.forName("org.drools.dynamic.DynamicComponentsSupplier", false, classLoader);
+            } catch (ClassNotFoundException e) {
+                throw new IllegalStateException(
+                        "Constraints DRL has been detected, but drools-core-dynamic is not on classpath."
```

Comment:
We put a newline between the "error fact" and the "error maybe suggestion":
```suggestion
                        "Using scoreDRL in Quarkus, but the dependency drools-core-dynamic is not on the classpath.\n"
```